lib/errors.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+class TldrError extends Error {
+  constructor(message) {
+    super(message);
+    this.name = 'TldrError';
+  }
+
+  static isTldrError(err) {
+    return err instanceof this;
+  }
+}
+
+class EmptyCacheError extends TldrError {
+  constructor() {
+    super(trim`
+Local cache is empty
+Please run tldr --update
+    `);
+    this.name = 'EmptyCacheError';
+    // eslint-disable-next-line no-magic-numbers
+    this.code = 2;
```

Comment:
What is the purpose of having error codes in the class object?

Reply:
To reuse it as exit code as spec mandates different exit codes for different fatal errors.